libs/core/langchain_core/language_models/chat_models.py

```diff
@@ -100,6 +102,26 @@ async def agenerate_from_stream(
     )
 
 
+def _as_async_iterator(sync_iterator: Callable) -> Callable:
+    """Convert a sync iterator into an async iterator."""
+
+    async def _as_sync_iterator(*args: Any, **kwargs: Any) -> AsyncGenerator:
+        iterator = await run_in_executor(None, sync_iterator, *args, **kwargs)
+        done = object()
+        while True:
+            doc = await run_in_executor(
```

Comment:
ooc what's doc stand for

Reply:
do copy pasta = latin for element or item